src/models/campaign.ts

```diff
@@ -154,6 +156,8 @@ export default class Campaign {
             .mapValues("dataElements")
             .value();
 
+        if (!vaccinationAttribute) return { status: false, error: "Metadata not found: vaccinationAttributeId" };
```

Comment:
You can add this to the if/else if/else chain. As a rule of thumb, we try to avoid early returns, create conditional branches, each with its return value.